keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import layers
+from keras_cv.utils import preprocessing
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class AugMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Performs the AugMix data augmentation technique.
+
+    AugMix mixes several chains of augmentations together using weights sampled
+    from a Dirichlet distribution.  A single chain consist of a series of
+    individual augmentations.  The resultant image is further mixed with the
+    original image to form the final augmented image.
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written (low, high).
+            This is typically either `(0, 1)` or `(0, 255)` depending
+            on how your preprocessing pipeline is setup.
+        severity: A tuple of two floats, a single float or a `keras_cv.FactorSampler`.
+            A value is sampled from the provided range.  If a float is passed, the
+            range is interpreted as `(0, severity)`. This value represents the
+            level of strength of augmentations and is in the range [0, 1].
+            Defaults to 0.3.
+        width: an integer representing the number of different chains to
```

Comment:
perhaps instead of `width` we can come up with a more descriptive name.

`chains`, `num_chains`, something of the sort?  Thoughts?

Reply:
I think I like num_chains.

Reply:
Do we want this changed from the original implementation and the paper terminology? 

I do agree with you on this however, this would make it more descriptive.